freqtrade/exchange/exchange.py

```diff
@@ -677,33 +677,64 @@ def price_get_one_pip(self, pair: str, price: float) -> float:
         else:
             return 1 / pow(10, precision)
 
-    def get_min_pair_stake_amount(self, pair: str, price: float, stoploss: float,
-                                  leverage: Optional[float] = 1.0) -> Optional[float]:
+    def get_min_pair_stake_amount(
+        self,
+        pair: str,
+        price: float,
+        stoploss: float,
+        leverage: Optional[float] = 1.0
+    ) -> Optional[float]:
+        return self._get_stake_amount_limit(pair, price, stoploss, 'min', leverage)
+
+    def get_max_pair_stake_amount(
+        self,
+        pair: str,
+        price: float,
+        stoploss: float
```

Comment:
I think stoploss should be hard-set to 0 for the max-case (and not provided to `get_max_stake_amount` at all).

the reason to have stoploss part of this method is the following:
if you place a buy order near the very bottom (say - for 11$ with 10$ as minimum limit)  - and then have stoplos 20% - that stop would mean a sell for 9$ - which the exchange will no longer allow as it's below minimium limits

at the top however, this is not relevant (if you bought 1_000_000 - you can still sell after a 20% drop (even though you should've sold earlier, but that's not freqtrade's problem).

Reply:
> I think stoploss should be hard-set to 0 for the max-case (and not provided to get_max_stake_amount at all).

A stoploss of 0 isn't correct for shorts, if it doesn't matter though I can just set it to that

> at the top however, this is not relevant (if you bought 1_000_000 - you can still sell after a 20% drop (even though you should've sold earlier, but that's not freqtrade's problem).

![](https://i.imgur.com/nTAaNOt.gif)

Reply:
> A stoploss of 0 isn't correct for shorts

It's not about shorts/not shorts - but about stoploss values per se for the upper bound.

Max defines the upper limit (in your example, 5000). It does not matter if your stoploss is 5% or 50% - the upper bound will still be 5000. This is different to the behavior on the lower bound, where we factor in the stoploss to be able to  exit the trade (see example above).

If you carefully look at the implementation for max -  you'll notice that stoploss is used for calculations - but the result of the calculation is ignored in case of the "max" calculation. It's therefore only relevant for "min" - which is correct (but makes the stoploss parameter irrelevant to calls that use "max" as "limit".

Reply:
I know that the `max_reserve_percent` isn't used for the max amount, but I don't know if that's correct behaviour

Reply:
if any, it'd be ROI that's needed, not stop 
Say we have a coin that trades from 10$ to 5000$ per order ...
we enter at minimum 15$ - and at maximum 5000$ ...
the problem with the 5k is - if it goes in our favor (it's now worth 5500$) - we might not be able to sell it anymore ...

long/short is not really relevant i think - as it's stake-currency we're talking here (mainly)